### PR TITLE
Add amount to item info when item is counted by charges

### LINKED ISSUE #21923: Add Crafting Result Amounts
Would it be possible to add resulting amounts to the crafting menu?  I keep ending up guessing batch sizes when making supplies for stuff.

### Expected behavior
Food tells you how many portions it makes:
cookie
I want other things to do the same. Example:
bat

### Actual behavior
No indication of created amounts:
image
Batteries I know are 100 but there are a lot of things I don't regularly craft.

### PR DESCRIPTION
**Pull request by laeaton2001**
Fixes #21923 
Adds an amount line to item info when the item is counted by charges & is not a food item (since they already have the portions info). This will show the resulting amount from any crafting.
It does however also show this info in every other item info screen so not 100% sure this is the best way to do this.
Some screenshots:
Crafting:
image
Item info:
image
Pickup:
image

### CODE REVIEW

**Comment by angoddu** [comment]
Maybe you could look at how the portions handles that and simply mimic it

**Comment by Coolthulhu** [comment]
* Portions and amount are displayed in different place. Either move portions up, delete them altogether, or move amount down to where portions would be.
* Batch crafting doesn't properly multiply amount. This isn't trivial to implement, so I guess it's not mandatory.

**Comment by laeaton2001** [comment]
Moved the data down to be where the portions data is for food. From what I can see batch crafting doesn't update any info except the requirements, the item info (weight, volume, etc.) remain static. I'm happy to look at that but think it'd be better in a separate PR.
I must have misunderstood 19764.

**Comment by laeaton2001** [comment]
Since this was hanging around I've added the multiplying of item info for batch size.

File: src/crafting_gui.cpp [hunk 1]
```diff
@@ -513,7 +513,12 @@ const recipe *select_crafting_recipe( int &batch_size )
                 if( last_recipe != current[line] ) {
                     last_recipe = current[line];
                     tmp = current[line]->create_result();
-                    tmp.info( true, thisItem );
+                }
+                if( batch ) {
```
Comment (BevapDin) [inline src/crafting_gui.cpp]: Both branches contain the very same code, why have a branch here at all?

Comment (laeaton2001) [reply src/crafting_gui.cpp]: My bad, this was a left over from testing. Fixed.

**Pull request merged**